src/odemis/driver/autoscript_client.py

```diff
@@ -0,0 +1,2304 @@
+"""
+
+class FIBSEM(model.HwComponent):
+    """
+    Driver to communicate with autoscript software on TFS microscopes. autoscript is the software TFS uses to control their microscopes.
+    To use this driver the autoscript adapter developed by Delmic should be running on the TFS PC. Communication to the
+    Microscope server is done via Pyro5.
+    """
+
+    def __init__(self, name, role, children, address, daemon=None,
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        address: str
+            server address and port of the Microscope server, e.g. "PYRO:Microscope@localhost:4242"
+        timeout: float
+            Time in seconds the client should wait for a response from the server.
+        """
+
+        # print versions of msgpack, msgpack-numpy, pyro
+        print("msgpack version: %s", pkg_resources.get_distribution("msgpack").version)
+        print("msgpack-numpy version: %s", pkg_resources.get_distribution("msgpack-numpy").version)
+        print("Pyro5 version: %s", pkg_resources.get_distribution("Pyro5").version)
+        print("Pyro4 version: %s", pkg_resources.get_distribution("Pyro4").version)
+
+        model.HwComponent.__init__(self, name, role, daemon=daemon, **kwargs)
+        self._proxy_access = threading.Lock()
+        try:
+            self.server = Pyro5.api.Proxy(address)
+            self.server._pyroTimeout = 30  # seconds
+            self._swVersion = self.server.get_software_version()
+            self._hwVersion = self.server.get_hardware_version()
+            logging.debug(
+                f"Successfully connected to autoscript server with software version {self._swVersion} and hardware"
+                f"version {self._hwVersion}")
+        except CommunicationError as err:
+            raise HwError("Failed to connect to autoscript server '%s'. Check that the "
+                          "uri is correct and autoscript server is"
+                          " connected to the network. %s %s" % (address, err, info))
+        except OSError as err:
+            raise HwError("XT server reported error: %s." % (err,))
+
+        # Transfer latest xtadapter package if available
+        # The transferred package will be a zip file in the form of bytes
+        # self.check_and_transfer_latest_package()
+
+        # Create the scanner type child(ren)
+        # Check if at least one of the required scanner types is instantiated
+        scanner_types = ["scanner", "fib-scanner"]  # All allowed scanners types
+        if not any(scanner_type in children for scanner_type in scanner_types):
+            raise KeyError("FIBSEM was not given any scanner as child. "
+                           "One of 'scanner', 'fib-scanner' or 'mb-scanner' need to be included as child")
+
+        has_detector = "detector" in children
+        has_fib_detector = "fib-detector" in children
+
+        if "scanner" in children: # TODO: rename to sem-scanner
```

Comment:
Please do the TODO.

Reply:
changed all components to sem-* or fib-*